Engine_Revit_UI/Query/ElementIds/ElementIdsOfFamilies.cs

```diff
@@ -0,0 +1,95 @@
+ * (at your option) any later version.                                          
+ *                                                                              
+ * The BHoM is distributed in the hope that it will be useful,              
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of               
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE. See the                 
+ * GNU Lesser General Public License for more details.                          
+ *                                                                            
+ * You should have received a copy of the GNU Lesser General Public License     
+ * along with this code. If not, see <https://www.gnu.org/licenses/lgpl-3.0.html>.      
+ */
+
+using System;
+using System.Linq;
+using System.Reflection;
+using System.Collections.Generic;
+using System.ComponentModel;
+
+using Autodesk.Revit.DB;
+using Autodesk.Revit.UI;
+
+using BH.oM.Base;
+using BH.Engine.Adapters.Revit;
+using BH.oM.Adapters.Revit;
+using BH.oM.Adapters.Revit.Enums;
+using BH.oM.Adapters.Revit.Interface;
+using BH.oM.Data.Requests;
+using BH.oM.Reflection.Attributes;
+
+namespace BH.UI.Revit.Engine
+{
+    public static partial class Query
+    {
+        /***************************************************/
+        /****              Public methods               ****/
+        /***************************************************/
+
+        [Description("Get the ElementId of all Families, with option to narrow search by Family name only")]
+        [Input("document", "Revit Document where ElementIds are collected")]
+        [Input("familyName", "Optional, narrows the search by a Family name. If blank it returns all Families")]        
+        [Input("caseSensitive", "Optional, sets the Family name and Family Type name to be case sensitive or not")]
+        [Input("ids", "Optional, allows the filter to narrow the search from an existing enumerator")]
+        [Output("elementIdsOfFamilies", "An enumerator for easy iteration of ElementIds collected")]
+        public static IEnumerable<ElementId> ElementIdsOfFamilies(this Document document, string familyName = null, bool caseSensitive = true, IEnumerable<ElementId> ids = null)
+        {
+            if (document == null)
+                return null;
+
+            if (!string.IsNullOrEmpty(familyName))
+            {
+                HashSet<ElementId> result = new HashSet<ElementId>();
+
+                Element element = null;
+
+                if (caseSensitive)
+                {
+                    element = new FilteredElementCollector(document).OfClass(typeof(Family)).Where(x => x.Name == familyName).FirstOrDefault();                    
+                }
+                else
+                {
+                    element = new FilteredElementCollector(document).OfClass(typeof(Family)).Where(x => x.Name.ToUpper() == familyName.ToUpper()).FirstOrDefault();                    
```

Comment:
Why not use case insensitive string equality checks? Not asking for a change, just curious :smile: